File: HRPayrollSystem/Frontend/src/types/auditLog.ts
export interface AuditLog {
  auditId: number;
  employeeId: string;
  action: 'Created' | 'Updated' | 'Deleted' | 'Login' | 'Logout';
  tableName: string;
  recordId: string;
  oldValues?: string;
  newValues?: string;
  timestamp: string;
  timeAgo: string;
}

export interface AuditLogFilters {
  search: string;
  employeeId: string;
  action: string;
  tableName: string;
  fromDate: string;
  toDate: string;
}

File: HRPayrollSystem/Backend/Models/AuditLog.cs
using System.ComponentModel.DataAnnotations;
using HRPayrollSystem_Payslip.Enums;

namespace HRPayrollSystem_Payslip.Models
{
    public class AuditLog
    {
        [Key]
        [Required]
        public int AuditId { get; set; }

        [Required]
        public string EmployeeId { get; set; }

        [Required]
        public ActionType Action { get; set; }

        [Required]
        public string TableName { get; set; }

        [Required]
        public string RecordId { get; set; }

        public string? OldValues { get; set; }

        public string? NewValues { get; set; }

        [Required]
        public DateTime Timestamp { get; set; } = DateTime.UtcNow;
    }

}
